addons/mail/static/src/js/composers/basic_composer.js

```diff
@@ -419,33 +529,7 @@ var BasicComposer = Widget.extend({
      * @param {jQuery.Event} ev
      */
     _onAttachmentChange: function (ev) {
-        var self = this;
-        var files = ev.target.files;
-        var attachments = this.get('attachment_ids');
-
-        _.each(files, function (file){
-            var attachment = _.findWhere(attachments, {name: file.name});
-            // if the files already exits, delete the file before upload
-            if (attachment){
-                self._attachmentDataSet.unlink([attachment.id]);
-                attachments = _.without(attachments, attachment);
-            }
-        });
-
-        this.$('form.o_form_binary_form').submit();
-        this._$attachmentButton.prop('disabled', true);
-        var uploadAttachments = _.map(files, function (file){
-            return {
-                id: 0,
-                name: file.name,
-                filename: file.name,
-                url: '',
-                upload: true,
-                mimetype: '',
-            };
-        });
-        attachments = attachments.concat(uploadAttachments);
-        this.set('attachment_ids', attachments);
+        this._processAttachmentChange({files: ev.target.files, submitForm: true});
```

Comment:
`target` => `currentTarget` + one line per property for object created on-the-fly:
```js
this._processAttachmentChange({
   files: ev.currentTarget.files,
   submitForm: true,
})
```